lib/mimdb_web/controllers/movie_controller.ex

```diff
@@ -0,0 +1,69 @@
+defmodule MimdbWeb.MovieController do
+  use MimdbWeb, :controller
+
+  alias Mimdb.Movies
+  alias Mimdb.Movies.Movie
+
+  def index(conn, _params) do
+    movies = Movies.list_movies()
+    render(conn, "index.html", movies: movies)
+  end
+
+  def new(conn, _params) do
+    changeset = Movies.change_movie(%Movie{})
+    genres = Movies.list_genres()
+    render(conn, "new.html", changeset: changeset, genres: genres)
+  end
+
+  def create(conn, %{"movie" => movie_params}) do
+    genres = Movies.list_genres()
+
+    case Movies.create_movie(movie_params) do
+      {:ok, movie} ->
+        conn
+        |> put_flash(:info, "Movie created successfully.")
+        |> redirect(to: Routes.movie_path(conn, :show, movie))
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        render(conn, "new.html", changeset: changeset, genres: genres)
+    end
+  end
+
+  def show(conn, %{"id" => id}) do
+    movie = Movies.get_movie!(id)
+    render(conn, "show.html", movie: movie)
+  end
+
+  def edit(conn, %{"id" => id}) do
+    movie = Movies.get_movie!(id)
+    changeset = Movies.change_movie(movie)
+    genres = Movies.list_genres()
+    roles = Movies.list_roles()
```

Comment:
I think this is a bug, the one Laura referenced

Reply:
Ah sure both the genres and roles need to be limited to just the ones for the movie.